antarest/study/storage/variantstudy/model/command/update_link.py

```diff
@@ -0,0 +1,83 @@
+# Copyright (c) 2024, RTE (https://www.rte-france.com)
+#
+# See AUTHORS.txt
+#
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+#
+# SPDX-License-Identifier: MPL-2.0
+#
+# This file is part of the Antares project.
+import typing as t
+
+from pydantic import ValidationError
+
+from antarest.core.exceptions import LinkValidationError
+from antarest.study.storage.rawstudy.model.filesystem.config.model import FileStudyTreeConfig
+from antarest.study.storage.rawstudy.model.filesystem.factory import FileStudy
+from antarest.study.storage.variantstudy.model.command.common import CommandName, CommandOutput
+from antarest.study.storage.variantstudy.model.command.create_link import AbstractLinkCommand, LinkProperties
+from antarest.study.storage.variantstudy.model.command.icommand import MATCH_SIGNATURE_SEPARATOR, ICommand, OutputTuple
+from antarest.study.storage.variantstudy.model.model import CommandDTO
+
+
+class UpdateLink(AbstractLinkCommand):
+    """
+    Command used to update a link between two areas.
+    """
+
+    # Overloaded metadata
+    # ===================
+
+    command_name: CommandName = CommandName.UPDATE_LINK
+    version: int = 1
+
+    def _apply_config(self, study_data: FileStudyTreeConfig) -> OutputTuple:
+        area_from, area_to = sorted([self.area1, self.area2])
+
+        return (
+            CommandOutput(
+                status=True,
+                message=f"Link between '{self.area1}' and '{self.area2}' updated",
+            ),
+            {"area_from": area_from, "area_to": area_to},
+        )
+
+    def _apply(self, study_data: FileStudy, *a: str) -> CommandOutput:
+        version = study_data.config.version
+        area_from, area_to = sorted([self.area1, self.area2])
+
+        try:
+            properties = LinkProperties.model_validate(self.parameters or {}).model_dump(
+                exclude_unset=True, by_alias=True
+            )
+        except ValidationError:
+            raise LinkValidationError("One or more fields are forbidden")
+
+        current_parameters = study_data.tree.get(["input", "links", area_from, "properties", area_to])
+
+        current_parameters.update(properties or {})
+
+        study_data.tree.save(current_parameters, ["input", "links", area_from, "properties", area_to])
+
+        output, _ = self._apply_config(study_data.config)
+
+        if any([self.series, self.direct, self.indirect]):
+            self.save_series(area_from, area_to, study_data, version)
```

Comment:
It's better cause you solved a part of the issue  I explained but I believe there's still an issue.
If someone just passes the series matrix inside the update you shouldn't modify the 2 others.
You can add a test to ensure this behaviour is correct